client/components/overviewWidget/ProductInformation.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import StarAverage from '../shared/StarAverage.jsx';
+// import axios from 'axios';
+
+class ProductInformation extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      product: null,
+      reviewMetaData: null,
+      reviewAverage: null,
+      reviewCount: 0
+    };
+  }
+
+  componentDidMount() {}
+
+  componentDidUpdate(prevProps) {
+    if (this.props.product !== prevProps.product) {
+      let ratings = this.props.reviewMetaData.ratings;
+      let totalRatings = 0;
+      for (let rating in ratings) {
+        totalRatings += parseInt(ratings[rating]);
+      }
+      this.setState({
```

Comment:
Flagging again that you are certain you need to do this since it is discouraged by React to set the state from props.